[FIX] payroll: show code field on hr.leave.type view

ivancandelas opened the pull request:
Impacted versions:

Odoo15 Community

Steps to reproduce:

go to menu Time Off Types and create a new record, the payroll code field won't show off

Expected behavior:

go to menu Time Off Types and create a new record, it will show the payroll code field

Description of the issue/feature this PR addresses:

Adds the field payroll code in the hr_leave_type_view_form view

ivancandelas committed:
[FIX] payroll: show code field on hr.leave.type view

mostafabarmshory commented:
Dear @ivancandelas 
You must run the pre-commit command locally and then commit the result.

The pre-commit command changes indents and fixes some issues.

nimarosa commented:
Hello, the fix is okay. I missed this when approving the migration. 

For this to be merged please fix pre-commit errors and we merge.

nimarosa commented:
@ivancandelas Please do not update dotfiles inside a PR. Anyways the current dotfiles are okay and will be updated to new versions in the future. 

The pre-commit error is related to a missing "," in the manifest. Please run pre-commit before you commit to the branch and this errors will be automatically solved by pre-commit. 

Please remove the last commit.

ivancandelas committed:
[FIX] Correct manifest typo

ivancandelas commented:
@nimarosa thank you I think i screwed up, so I updated the dotfiles because i was getting the following runtime error:

 RuntimeError: The Poetry configuration is invalid:
        - [extras.pipfile_deprecated_finder.2] 'pip-shims<=0.3.4' does not match '^[a-zA-Z-_.0-9]+$'

so changing the dotfiles fixed the error and the I realize the missing "," in the manifest.

Then, I need to revert the dotfiles commit?

ivancandelas commented:
@nimarosa thank you, let me remove the commit that changed dotfiles

ivancandelas commented:
I removed the dotfiles commit, also fixed the manifest, still the pre-commit check is not successful:

  RuntimeError: The Poetry configuration is invalid:
            - [extras.pipfile_deprecated_finder.2] 'pip-shims<=0.3.4' does not match '^[a-zA-Z-_.0-9]+$'

With the dotfiles commit and the manifest fixed the pre-commit check was successful

what I'm doing wrong?

nimarosa commented:
@ivancandelas let me rerun the job to see what is happening and I get back to you.

nimarosa commented:
@ivancandelas Okay so apparently we have to update dotfiles. Weird error but new dotfiles version fixes it so we are going to do it. 

I will do it in another commit and then rebase this PR and we should see everything in green. When it's done I merge this.

ivancandelas commented:
@nimarosa the latest release of [oca-addons-repo-template](https://github.com/OCA/oca-addons-repo-template/releases/tag/v1.14.2)  has the fix for the pre-commit issues

nimarosa commented:
/ocabot rebase

nimarosa commented:
Dotfiles updated, tests passing merging. 

/ocabot merge patch.
